libs/owasp-diff-runner/src/main/java/ddf/lib/OwaspDiffRunner.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.lib;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.InvocationResult;
+import org.apache.maven.shared.invoker.Invoker;
+import org.apache.maven.shared.invoker.MavenInvocationException;
+import org.codehaus.plexus.util.StringUtils;
+
+public class OwaspDiffRunner {
+
+    public static final String MAVEN_VERSION_COMMAND = "mvn -version";
+
+    public static final String MAVEN_SETTINGS_COMMAND = "mvn help:effective-settings";
+
+    public static final String GIT_DIFF_NAME_COMMAND = "git diff --name-only ";
+
+    public static final String GIT_SHORT_BRANCH_NAME_COMMAND = "git rev-parse --abbrev-ref HEAD";
+
+    private static final String DASHES = StringUtils.repeat("-", 25);
+
+    public static final String PROJECT_ROOT = System.getProperty("user.dir")
+            .split("/libs/owasp-diff-runner")[0];
+
+    private static String mavenHome;
+
+    private static String localRepo;
+
+    private static Runtime runTime = Runtime.getRuntime();
+
+    private static Invoker invoker = new DefaultInvoker();
+
+    public OwaspDiffRunner(Runtime runTime, Invoker invoker) {
+        this.runTime = runTime;
+        this.invoker = invoker;
+    }
+
+    public static void main(String[] args) throws OwaspDiffRunnerException {
+
+        System.out.println(DASHES + " BEGIN OWASP DIFF AUDIT: " + DASHES);
+        mavenHome = getMavenHome();
+        localRepo = getLocalRepo();
+        String modulesOfChangedPoms = getModulesOfChangedPoms();
+        if (modulesOfChangedPoms.isEmpty()) {
+            System.out.println("No changed poms.");
+            return;
+        }
+
+        InvocationRequest request = new DefaultInvocationRequest();
+        request.setPomFile(new File(PROJECT_ROOT + File.separator + "pom.xml"));
+        request.setBaseDirectory(new File(PROJECT_ROOT));
+        request.setLocalRepositoryDirectory(new File(localRepo));
+
+        request.setGoals(Arrays.asList("dependency-check:check",
```

Comment:
I abstained from doing it since there is no version property for the owasp plugin. I would have had to change several poms to do so. It may be a good idea to do this when upgrade the plugin